src/components/carousel/carousel.js

```diff
@@ -92,11 +138,46 @@ class Carousel extends React.Component {
    * @method componentWillMount
    */
   componentWillMount() {
-    const selectedIndex = Number(this.props.initialSlideIndex);
+    const selectedIndex = Number(this.props.slideIndex) || Number(this.props.initialSlideIndex);
     this.setState({ selectedSlideIndex: selectedIndex });
   }
 
   /**
+   * A lifecycle method that is called before re-render.
+   *
+   * @method componentWillReceiveProps
+   */
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.slideIndex === undefined) { return; }
+
+    let newIndex = nextProps.slideIndex;
+    const currentIndex = this.state.selectedSlideIndex;
+
+    if (newIndex < 0) {
```

Comment:
this method is a bit cluttered - lines 156-162 could be pulled out into a method.
Possibly the same for line 172-177 (a separate method)